src/util/http_utils.js

```diff
@@ -719,8 +719,8 @@ function set_cors_headers_s3(req, res, cors_rules) {
         // https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html
         dbg.log0('set_cors_headers_s3: found matching CORS rule:', matched_rule);
         set_cors_headers(req, res, {
-            allow_origin: matched_rule.allowed_origins.includes('*') ? '*' : req.headers.origin,
-            allow_methods: matched_rule.allowed_methods.join(','),
+            allow_origin: match_origin,
+            allow_methods: match_method,
```

Comment:
This is also not correct - take a look at the above example
Allow-Methods in the rule are: GET, POST, PUT, DELETE
, and match_method is GET

Reply:
I agree, this was not the correct solution to the problem. 
Will revert back. I have found more issues and will discuss it with you.